src/base/ERC4626Proxy.sol

```diff
@@ -0,0 +1,201 @@
+            uint256 allowed = allowance[owner][msg.sender]; // Saves gas for limited approvals.
+            if (allowed != type(uint256).max) allowance[owner][msg.sender] = allowed - shares;
+        }
+
+        // Burn
+        _burn(owner, shares);
+
+        // Redeem and emit
+        assets = underlyingVault.redeem(shares, receiver, address(this));
+        emit Withdraw(msg.sender, receiver, owner, assets, shares);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            ACCOUNTING LOGIC
+    //////////////////////////////////////////////////////////////*/
+
+    function totalAssets() public view virtual returns (uint256) {
+        return convertToAssets(underlyingVault.balanceOf(address(this)));
+    }
+
+    function convertToShares(uint256 assets) public view virtual returns (uint256) {
+        return underlyingVault.convertToShares(assets);
+    }
+
+    function convertToAssets(uint256 shares) public view virtual returns (uint256) {
+        return underlyingVault.convertToAssets(shares);
+    }
+
+    function previewDeposit(uint256 assets) public view virtual returns (uint256) {
+        return underlyingVault.previewDeposit(assets);
+    }
+
+    function previewMint(uint256 shares) public view virtual returns (uint256) {
+        return underlyingVault.previewMint(shares);
+    }
+
+    function previewWithdraw(uint256 assets) public view virtual returns (uint256) {
+        return underlyingVault.previewWithdraw(assets);
+    }
+
+    function previewRedeem(uint256 shares) public view virtual returns (uint256) {
+        return underlyingVault.previewRedeem(shares);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                     DEPOSIT/WITHDRAWAL LIMIT LOGIC
+    //////////////////////////////////////////////////////////////*/
+
+    function maxDeposit(address) public view returns (uint256) {
+        return underlyingVault.maxDeposit(address(this));
+    }
+
+    function maxMint(address) public view returns (uint256) {
+        return underlyingVault.maxMint(address(this));
+    }
+
+    function maxWithdraw(address owner) public view returns (uint256) {
+        uint256 maxWithdrawVault = underlyingVault.maxWithdraw(address(this));
+        uint256 assetsBalance = previewRedeem(balanceOf[owner]);
+        return maxWithdrawVault < assetsBalance ? maxWithdrawVault : assetsBalance;
+    }
+
+    function maxRedeem(address owner) public view returns (uint256) {
+        uint256 maxRedeemVault = underlyingVault.maxRedeem(address(this));
+        uint256 shareBalance = balanceOf[owner];
+        return maxRedeemVault < shareBalance ? maxRedeemVault : shareBalance;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                              AD HOC LOGIC
+    //////////////////////////////////////////////////////////////*/
+
+    function maxApproveUnderlyingVault() public {
```

Comment:
Just in case it needs to be called again. Shouldn't happen, but better safe than sorry